src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/UnsafeNativeMethodsCLR.cs

```diff
@@ -964,7 +964,8 @@ internal static void PostMessage(HandleRef hwnd, WindowMessage msg, IntPtr wpara
         {
             if (!IntPostMessage(hwnd, msg, wparam, lparam))
             {
-                throw new Win32Exception();
+                int errorCode = Marshal.GetLastWin32Error();
+                throw new Win32Exception(errorCode);
```

Comment:
```suggestion
                throw new Win32Exception();
```
This change is unnecessary, the empty ctor already calls `GetLastWin32Error` (technically, its `GetLastPInvokeError`, which is what `GetLastWin32Error` will call).

Reply:
@h3xds1nz yeah that is the case, but this is added here to avoid any other exception that the empty constructor call might return due to a different failure from any other thread (very less likely though). This follows the practice implemented in the same file. 

**Ref:** https://github.com/dotnet/wpf/blob/3a0442ac70cc6bbb15f45d3b60aa80f6758f234c/src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/UnsafeNativeMethodsCLR.cs#L396-L409

Reply:
@h3xds1nz thanks for this piece of information. Makes sense to not explicitly store the error if we are avoiding bad reads. Will go ahead and remove this part.

Reply:
@harshit7962 welcome